src/LibraryManager/LibrariesValidator.cs

```diff
@@ -24,141 +28,346 @@ internal class LibrariesValidator
         /// <param name="defaultProvider"></param>
         public LibrariesValidator(IDependencies dependencies, string defaultDestination, string defaultProvider)
         {
-            _dependencies = dependencies ?? throw new ArgumentNullException(nameof(dependencies));
+            _dependencies = dependencies;
```

Comment:
Is null allowed for _dependencies?

Reply:
No null is not allowed. But if _dependencies is null we would have failed way back. Callers should ensure that. Previous code was throwing and I would have to add try/catch all over the place for something that should never happen. 


---
In reply to: [195514946](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195514946) [](ancestors = 195514946)